neural_modelling/src/neuron/implementations/neuron_impl_stoc_exp.h

```diff
@@ -0,0 +1,250 @@
+    state->inputs[1] = ZERO;
+}
+
+static inline void neuron_model_save_state(neuron_impl_t *state, neuron_params_t *params) {
+	params->refract_init = state->refract_timer;
+	spin1_memcpy(params->random_seed, state->random_seed, sizeof(mars_kiss64_seed_t));
+}
+
+SOMETIMES_UNUSED // Marked unused as only used sometimes
+static void neuron_impl_load_neuron_parameters(
+        address_t address, uint32_t next, uint32_t n_neurons,
+        address_t save_initial_state) {
+
+    neuron_params_t *params = (neuron_params_t *) &address[next];
+    for (uint32_t i = 0; i < n_neurons; i++) {
+        neuron_model_initialise(&neuron_array[i], &params[i]);
+    }
+
+    // If we are to save the initial state, copy the whole of the parameters
+    // to the initial state
+    if (save_initial_state) {
+        spin1_memcpy(save_initial_state, address,
+        		n_neurons * sizeof(neuron_params_t));
+    }
+}
+
+SOMETIMES_UNUSED // Marked unused as only used sometimes
+static void neuron_impl_store_neuron_parameters(
+        address_t address, uint32_t next, uint32_t n_neurons) {
+    neuron_params_t *params = (neuron_params_t *) &address[next];
+	for (uint32_t i = 0; i < n_neurons; i++) {
+		neuron_model_save_state(&neuron_array[i], &params[i]);
+	}
+}
+
+SOMETIMES_UNUSED // Marked unused as only used sometimes
+static void neuron_impl_add_inputs(
+        index_t synapse_type_index, index_t neuron_index,
+        input_t weights_this_timestep) {
+    // Get the neuron itself
+    neuron_impl_t *neuron = &neuron_array[neuron_index];
+
+    // Do something to store the inputs for the next state update
+    neuron->inputs[synapse_type_index] += weights_this_timestep;
+}
+
+SOMETIMES_UNUSED // Marked unused as only used sometimes
+static void neuron_impl_do_timestep_update(
+        uint32_t timer_count, uint32_t time, uint32_t n_neurons) {
+    for (uint32_t neuron_index = 0; neuron_index < n_neurons; neuron_index++) {
+        // Get the neuron itself
+        neuron_impl_t *neuron = &neuron_array[neuron_index];
+
+        // If in refractory, count down and spike!
+		if (neuron->refract_timer > 0) {
+			neuron->refract_timer -= 1;
+
+			// Record things
+			neuron_recording_record_int32(PROB_INDEX, neuron_index, 0);
+			neuron_recording_record_accum(V_RECORDING_INDEX, neuron_index, ZERO);
+			neuron_recording_record_accum(EX_INPUT_INDEX, neuron_index, neuron->inputs[0]);
+			neuron_recording_record_accum(IN_INPUT_INDEX, neuron_index, neuron->inputs[1]);
+
+			// Reset the inputs
+			neuron->inputs[0] = ZERO;
+			neuron->inputs[1] = ZERO;
+
+			// Send a spike
+			neuron_recording_record_bit(SPIKE_RECORDING_BITFIELD, neuron_index);
+		    send_spike(timer_count, time, neuron_index);
+		    continue;
+		}
+
+        // Work out the membrane voltage
+        REAL v_membrane = (neuron->bias + neuron->inputs[0]) - neuron->inputs[1];
+
+        // Record things
+        neuron_recording_record_accum(V_RECORDING_INDEX, neuron_index, v_membrane);
```

Comment:
some timesteps will record twice.
yet neuron recorder sampling rates are 1
this will cause you to run out of sdram on a long run!